cmd/object-handlers.go

```diff
@@ -2293,7 +2293,12 @@ func (api objectAPIHandlers) DeleteObjectHandler(w http.ResponseWriter, r *http.
 		return
 	}
 
-	if s3Error := checkRequestAuthType(ctx, r, policy.DeleteObjectAction, bucket, object); s3Error != ErrNone {
+	replica := r.Header.Get(xhttp.AmzBucketReplicationStatus) == replication.Replica.String()
+	action := policy.DeleteObjectAction
+	if replica {
+		action = policy.ReplicateDeleteAction
+	}
+	if s3Error := checkRequestAuthType(ctx, r, policy.Action(action), bucket, object); s3Error != ErrNone {
```

Comment:
AFAICT nothing prevents someone from just setting this property and change the check?

It seems to me we really needs some documentation to make sure that replica users are never mixed with regular users.

Reply:
This change is just relaxing the policy requirement, currently it requires both DeleteObject() and ReplicateDelete()

For replicated request we only need ReplicateDelete() that's all

Reply:
Yeah, but my point is that you can change the permission required from `DeleteObjectAction` to `ReplicateDeleteAction` by setting a header in the request.

If users are somehow given `ReplicateDeleteAction`, a hostile user could delete objects without `DeleteObjectAction` permission.

Reply:
if that confusion can arise, I can simply remove this.